src/main/java/com/google/sps/data/DatastoreAvailabilityDao.java

```diff
@@ -147,10 +145,16 @@ public void deleteInRangeForUser(String email, long minTime, long maxTime) {
     return availability;
   }
 
-  private List<Entity> getEntitiesInRange(long minTime, long maxTime, Optional<Filter> filterOpt) {
+  private List<Entity> getEntitiesInRange(
+      Instant minTime, Instant maxTime, Optional<Filter> filterOpt) {
     Filter minFilter =
-        new FilterPredicate("startTime", FilterOperator.GREATER_THAN_OR_EQUAL, minTime);
-    Filter maxFilter = new FilterPredicate("startTime", FilterOperator.LESS_THAN_OR_EQUAL, maxTime);
+        new FilterPredicate(
+            "startTime", FilterOperator.GREATER_THAN_OR_EQUAL, minTime.toEpochMilli());
+    Filter maxFilter =
+        new FilterPredicate(
+            "startTime",
+            FilterOperator.LESS_THAN_OR_EQUAL,
+            maxTime.minus(15, ChronoUnit.MINUTES).toEpochMilli());
```

Comment:
l thought maxTime was the last _start_ time that was acceptable. So that means `FilterOperator.LESS_THAN_OR_EQUAL, maxTime.toEpochMilli()`

Reply:
Nope, the maximum time of the range, so the end of the last slot.